### ISSUE TITLE
Docker Start failed on nightlys build 7288

### ISSUE THREAD

**Issue by rajanashok** [vmware/vic#3431]
Log File Location:
[https://storage.cloud.google.com/vic-ci-logs/functional_logs_vic_7288.zip?authuser=1&_ga=1.210014037.1122363360.1480435214](url)

Robot Log:

Running command 'docker -H sc-rdops-vm04-dhcp-109-114.eng.vmware.com:2376 --tls start 138707d442ee2b01cb8a534b8604c57e82aa93e8fe371b4951f8d5adc15cdd48 2>&1'.

${rc} = 1

${output} = Error response from daemon: Server error from portlayer: Get http://127.0.0.1:2377/containers/138707d442ee2b01cb8a534b8604c57e82aa93e8fe371b4951f8d5adc15cdd48: dial tcp 127.0.0.1:2377: getsockopt: con...

PortLayer Log:

```time=2016-12-02T08:20:23.197615894Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/exec.Commit:35] 138707d442ee2b01cb8a534b8604c57e82aa93e8fe371b4951f8d5adc15cdd48 
time=2016-12-02T08:20:32.535414257Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/exec.Commit:35] [9.337793326s] 138707d442ee2b01cb8a534b8604c57e82aa93e8fe371b4951f8d5adc15cdd48 
time=2016-12-02T08:20:32.547952320Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).CommitHandler:223] [9.352692274s] handle(d2814e2d12f849afd84664501ce76c34) 
time="2016-12-02T08:20:32Z" level=debug msg="Found 1 subscribers to events.ContainerEvent: Container 138707d442ee2b01cb8a534b8604c57e82aa93e8fe371b4951f8d5adc15cdd48 Created" 
time="2016-12-02T08:20:32Z" level=debug msg="Event manager calling back to netCtx(0xc42000f2c0)" 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xcdfd17]

goroutine 18 [running]:
panic(0xec2240, 0xc420012050)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/vmware/vic/vendor/github.com/vmware/govmomi/event.(*eventTailer).newEvents(0xc42032490c, 0xc4205e8500, 0xa, 0x10, 0x1, 0x0, 0xe95460)
	/drone/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/event/processor.go:176 +0x57
github.com/vmware/vic/vendor/github.com/vmware/govmomi/event.(*eventProcessor).process(0xc4206a9da8, 0xc42026afe0, 0x15, 0xc420701260, 0x51, 0xc42092d080, 0x1, 0x4, 0x11, 0xc4208bea20)
	/drone/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/event/processor.go:144 +0x215
github.com/vmware/vic/vendor/github.com/vmware/govmomi/event.(*eventProcessor).run.func2(0xc42092d080, 0x1, 0x4, 0x0)
	/drone/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/event/processor.go:119 +0x9a
github.com/vmware/vic/vendor/github.com/vmware/govmomi/property.Wait.func1(0xc420112a60, 0x15, 0xc42000ed80, 0x51, 0xc42092d080, 0x1, 0x4, 0x0)
	/drone/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/property/wait.go:69 +0x43
github.com/vmware/vic/vendor/github.com/vmware/govmomi/property.waitLoop(0x1abe6e0, 0xc420012458, 0xc420202d20, 0xc4206a9898, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/property/wait.go:142 +0x1e5
github.com/vmware/vic/vendor/github.com/vmware/govmomi/property.Wait(0x1abe6e0, 0xc420012458, 0xc4206a9a50, 0xc42026afe0, 0x15, 0xc420701260, 0x51, 0xc4203249a0, 0x1, 0x1, ...)
	/drone/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/property/wait.go:70 +0x30a
github.com/vmware/vic/vendor/github.com/vmware/govmomi/event.(*eventProcessor).run(0xc4206a9da8, 0x1abe6e0, 0xc420012458, 0xc42077b601, 0x16, 0xc42077b637)
	/drone/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/event/processor.go:128 +0x90d
github.com/vmware/vic/vendor/github.com/vmware/govmomi/event.Manager.Events(0x0, 0x0, 0xc4200ea2c0, 0xc42066dd10, 0xc, 0xc42066dd40, 0xc, 0xc4203456e0, 0xc42075d6b8, 0xa, ...)
	/drone/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/event/manager.go:175 +0x1e0
github.com/vmware/vic/lib/portlayer/event/collector/vsphere.(*EventCollector).Start.func1(0x1abe6e0, 0xc420012458, 0x100000001, 0x1, 0xc42077b660, 0x1, 0x1, 0xc420345740, 0xc42066dd40, 0xc)
	/drone/src/github.com/vmware/vic/lib/portlayer/event/collector/vsphere/collector.go:139 +0xfd
created by github.com/vmware/vic/lib/portlayer/event/collector/vsphere.(*EventCollector).Start
	/drone/src/github.com/vmware/vic/lib/portlayer/event/collector/vsphere/collector.go:146 +0x316
time="2016-12-02T08:20:38Z" level=info msg="Launching portlayer server pprof server on 127.0.0.1:6063" 
time="2016-12-02T08:20:38Z" level=debug msg="secret key decoded from guestinfo.ovfEnv" 
time="2016-12-02T08:20:38Z" level=debug msg="Session Environment Info: " API Type=VirtualCenter API Version=6.5 Build=4602587 Name="VMware vCenter Server" OS Type=linux-x64 Product ID=vpx UUID=4315cc8a-78df-4b33-94f6-bc010ea77dde Vendor="VMware, Inc." Version=6.5.0```

**Comment by dougm** [comment]
The govmomi event code hasn't been touched recently.  @hmahmood and @cgtexmex can you take a look?

```
% git blame -L 176,176 event/processor.go
a9cee43a (Hasan Mahmood 2016-08-23 13:50:01 -0500 176)                  if evs[i].GetEvent().Key != t.lastKey {
```

**Comment by dougm** [comment]
@rajanashok are you able to reproduce this?

**Comment by rajanashok** [comment]
This happened on both VSAN tests on the nightly, so I am assuming it is.

**Comment by anchal-agrawal** [comment]
For the `5-6-1-VSAN-Simple` test, I see this error message in `vic-machine.log`:
```
[34mINFO[0m[2016-12-02T07:40:27Z] Uploading images for container               
[34mINFO[0m[2016-12-02T07:40:27Z] 	"bin/bootstrap.iso"                         
[34mINFO[0m[2016-12-02T07:40:27Z] 	"bin/appliance.iso"                         
[31mERRO[0m[2016-12-02T07:40:28Z] 		Upload failed for "bin/bootstrap.iso": Put https://10.160.8.143/folder/5f254158-9dbf-bd9d-8eca-020001208305/V0.8.0-RC3-7288-95903A8-bootstrap.iso?dcPath=%2FvcqaDC&dsName=vsanDatastore: write tcp 172.17.0.3:50130->10.160.8.143:443: write: connection reset by peer 
[33mWARN[0m[2016-12-02T07:40:28Z] 		Continuing despite failures (due to --force option) 
[33mWARN[0m[2016-12-02T07:40:28Z] 		Note: The VCH will not function without "bin/bootstrap.iso"... 
```

However, this didn't happen in the other test (`5-6-2-VSAN-Complex`) where we saw the `docker start` failure.

**Comment by rajanashok** [comment]
I am re-running the VSAN Complex to get a repro now.

**Comment by dougm** [comment]
If anyone wants a nimbus vsan cluster without using robot to setup:

```bash
nimbus-testbeddeploy --noSupportBundles --vcvaBuild 4602587 --esxPxeDir 4564106 --esxBuild 4564106 --testbedName vcqa-vsan-simple-pxeBoot-vcva --runName $USER-vic-vsan --resultsDir nimbus-vsan

<./nimbus-vsan/testbedInfo.json jq -r '.vc | map(.vimUsername), map(.vimPassword), map(.ip4) | join(" ")' | xargs printf "export GOVC_URL='%s:%s@%s'\n"

$GOPATH/src/github.com/vmware/vic/infra/machines/vcsa/create-cluster.sh -d vcqaDC -c cls $(govc env -x GOVC_URL_HOST) $(govc ls -t HostSystem host/* | xargs -n1 basename)
```

**Comment by dougm** [comment]
I haven't run the robot tests against the cluster above, but basic start works for me:

```bash
docker pull busybox
id=$(docker create busybox /bin/top)
docker start $id
docker logs $id
docker ps
docker stop $id
```

**Comment by hickeng** [comment]
@dougm from the failure logs, the portlayer is getting restarted and it may well be fine on the second time around.
You'll need to check the portlayer log for the panic, or deploy with --debug=3 to suppress component relaunch

**Comment by hmahmood** [comment]
@dougm that panic is really strange. https://github.com/vmware/vic/blob/master/vendor/github.com/vmware/govmomi/event/processor.go#L176 ; the only possibility is that `GetEvent()` returns nil.

**Comment by rajanashok** [comment]
Have a repro of this issue on a nightly VSAN-Simple setup. Below are the details.

<img width="661" alt="screen shot 2016-12-02 at 3 14 44 pm" src="https://cloud.githubusercontent.com/assets/21340516/20850716/5b70ee52-b8a2-11e6-87d7-010665b06456.png">

**Comment by dougm** [comment]
@hickeng no panics in the port-layer.log in my env either

**Comment by dougm** [comment]
ok I can see what's going on using @rajanashok env.  We are getting a handful of events for a type that does not exist in govmomi (HostSubSpecificationUpdateEvent).  The event type is new to 6.5 and we started to receive them now that the API version defaults to 6.5.  The quick fix is setting it back to 6.0 (just a 1-liner), which is fine as we aren't using any 6.5 specific API yet.  I'll submit that change and revisit the phantom event type afterwards.

**Comment by dougm** [comment]
The HostSubSpecificationUpdateEvent type is internal, which is why there no govmomi type - not in the public wsdl, not generated.  VC shouldn't be sending us internal events, so this is most likely a 6.5
